luigi/task.py

```diff
@@ -575,6 +575,40 @@ def output(self):
         """
         return []  # default impl
 
+    def process_requires(self, requires):
+        """
+        Override in "template" tasks which themselves are supposed to be
+        subclassed and thus have their requires() overridden (name preserved to
+        provide consistent end-user experience), yet need to introduce
+        custom behaviour into dependencies.
+
+        Differences with :py:meth:`_requires`:
+          * method signature: :py:meth:`_requires` doesn't take params and calls
+            :py:meth:`requires` directly where as this method is expected to work only on
+            requirements passed via parameters.
+          * Code paths: :py:meth:`_requires` is run only during the statically defined
+            requirements where as this method works with dynamic dependencies yielded from
+            :py:meth:`run` also.
+
+        ex:
+            class ParentTask(luigi.Task):
+                foo = `bar`
+
+                def process_requires(self, requirements):
+                    for(req in flatten(requirements)):
+                        req.foo = self.foo
```

Comment:
Seem scary to modify input arguments like that. Not sure if the docs should encourage it.

Reply:
In our use case `foo` is actually used for log tags used to identify logs of a single pipeline throughout the run.
It's a powerful ability and I agree it could be dangerous to modify the input arguments and that it should be used with caution.

Reply:
Sure. Feel free to use it like that, but let's not encourage it in the docs?